docs/devguide/contributing.rst

```diff
@@ -35,3 +35,76 @@ negative votes, the team will grant commit privileges.
 The core team will also be responsible for removing commit privileges when
 appropriate - for example for malicious merge behavior or just inactivity over
 an extended period of time.
+
+
+Quality gates for the pull requests
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+There are three quality gates to ensure the pull requests quality, `Hound`_ for
+code style check, `Travis CI`_ for unit-test, `Jenkins`_ for re-unit-test and
+functional test. When a pull request created, all tests will run automatically, test 
+results can be found in the merge status field of each pull request page.
+
+.. _Hound: https://houndci.com/
+.. _Travis CI: https://travis-ci.org/
+.. _Jenkins: https://jenkins.io/
+.. _jshint: http://jshint.com/
+
+**Hound**
+
+Hound works with `jshint`_ and comments on style violations in pull requests. 
+Configuration files ``.hound.yml`` and ``.jshintrc`` have been created in each 
+repository, so before creating a pull request, you can check code style locally with 
+jshint to find out style violations beforehand.
+
+**Travis CI**
+
+Travis CI runs the unit tests, and then does some potentially ancillary actions.
+The build specifics are detailed in the ``.travis.yml`` file within each repository.
+For finding out basic errors before creating a pull request, you can run unit test
+locally using scripts ``HWIMO-TEST`` within each repository.
```

Comment:
Actually HWIMO-TEST does more than unit-testing, it still covers the npm package cleanup, re-installation and test coverage check. So to guide people do do unit-testing, the command `npm test` will be more suitable.